### PR TITLE
http: fix encodeError can't write response header

### LINKED ISSUE #779: http.Error 方法中 Response Header Content-Type 失效问题
transport/http包响应请求的`Accept`的代码片段如下
```
// encodeError encodes the erorr to the HTTP response.
func encodeError(w http.ResponseWriter, r *http.Request, err error) {
	se, ok := errors.FromError(err)
	if !ok {
		se = &errors.StatusError{
			Code:    2,
			Message: err.Error(),
		}
	}
	w.WriteHeader(se.HTTPStatus())
	encodeResponse(w, r, se)
}
```

然后在`encodeResponse`方法中通过匹配`Accept`字段返回对应的`Content-Type`
```
func encodeResponse(w http.ResponseWriter, r *http.Request, v interface{}) error {
	var codec encoding.Codec
	for _, accept := range r.Header[acceptHeader] {
		if codec = encoding.GetCodec(contentSubtype(accept)); codec != nil {
			break
		}
	}
	if codec == nil {
		codec = encoding.GetCodec("json")
	}
	data, err := codec.Marshal(v)
	if err != nil {
		return err
	}
	w.Header().Set(contentTypeHeader, contentType(codec.Name()))
	w.Write(data)
	return nil
}
```
通过Google发现`http.ResponseWriter`在调用`WriteHeader`方法之后的`Header().Set()`方法是无法设置响应头内容的，目前`http.Error`返回的`Content-Type`都是`text/plain`类型的

### PR DESCRIPTION
**Pull request by drrrMikado**
fix #779

### CODE REVIEW

File: transport/http/handle.go [hunk 1]
```diff
@@ -122,8 +115,25 @@ func encodeError(w http.ResponseWriter, r *http.Request, err error) {
...
+	codec := getCodec(r)
+	data, _ := codec.Marshal(se)
+	w.Header().Set(contentTypeHeader, contentType(codec.Name()))
 	w.WriteHeader(se.HTTPStatus())
-	encodeResponse(w, r, se)
+	_, _ = w.Write(data)
+}
+
+// getCodec get encoding.Codec via http.Request
+func getCodec(r *http.Request) encoding.Codec {
```
Comment (tonybase) [inline transport/http/handle.go:126]: codecForRequest(r *http.Request)

File: transport/http/handle.go [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package http
 
 import (
+	"github.com/go-kratos/kratos/v2/encoding/json"
```
Comment (tonybase) [inline transport/http/handle.go:4]: import (
    stdlib

    current_project

    third_party
)

File: transport/http/handle.go (hunk 1 above)
Comment (drrrMikado) [reply transport/http/handle.go:126]: Maybe `codecFromRequest` is better?

Comment (tonybase) [reply transport/http/handle.go:126]: https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#ReasonForError
https://pkg.go.dev/github.com/grpc-ecosystem/grpc-gateway/runtime#MarshalerForRequest

This method can be likes grpc gataway.

Comment (drrrMikado) [reply transport/http/handle.go]: Thanks

**Pull request merged**